### PR TITLE
Update Mlflow examples.

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
## What changes are proposed in this pull request?
 Update examples to work with mlflow 1.0 + some minor cleanup here and there. The biggest change is in multistep workflow example. 
(Please fill in changes proposed in this fix)
 
## How is this patch tested?
I manually ran all the examples.
### Is this a user-facing change? 

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [x] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/multistep_workflow/conda.yaml [hunk 1]
```diff
@@ -4,8 +4,9 @@ channels:
 dependencies:
   - python=3.6
   - pyspark
-  - keras
   - requests
   - click
   - pip:
-    - mlflow
+    - tensorflow==1.13.1
+    - keras==2.2.4
```
Comment (sueann) [inline examples/multistep_workflow/conda.yaml]: why did this need to be moved under pip do you know?

File: examples/multistep_workflow/main.py [hunk 2]
```diff
@@ -43,19 +44,21 @@ def _already_ran(entry_point_name, parameters, source_version, experiment_id=Non
             eprint(("Run matched, but is not FINISHED, so skipping "
                     "(run_id=%s, status=%s)") % (run_info.run_id, run_info.status))
             continue
-        if run_info.source_version != source_version:
+
+        if git_commit != tags.get(mlflow_tags.MLFLOW_GIT_COMMIT, None):
             eprint(("Run matched, but has a different source version, so skipping "
-                    "(found=%s, expected=%s)") % (run_info.source_version, source_version))
+                    "(found=%s, expected=%s)") % (run_info.source_version, git_commit))
```
Comment (sueann) [inline examples/multistep_workflow/main.py]: should it be `tags.get(mlflow_tags.MLFLOW_GIT_COMMIT, None)` instead of `run_info.source_version` ?

**Review by sueann — COMMENTED** [review]
just a few questions for you

File: examples/multistep_workflow/conda.yaml (hunk 1 above)
Comment (tomasatdatabricks) [reply examples/multistep_workflow/conda.yaml]: Good question I should have put a comment.  

The mkl version of tensorflow from conda and Spark don't like each other. There is a library that ends up loaded twice, complains and kills the program. You can get around by installing nomkl package from conda (seems shady) or setting an environment variable instructing the library to allow this behavior (not recommended by the authors). 

I think the safest and cleanest is to install from pip.

File: examples/multistep_workflow/main.py (hunk 2 above)
Comment (tomasatdatabricks) [reply examples/multistep_workflow/main.py]: Good catch! Yes it should.

**Review by sueann — APPROVED** [review]
lgtm - thanks!

**Pull request merged**